octopusdeploy_variables data source is not usable?

**Describe the bug**
The `octopusdeploy_variables` data source does not appear to be usable. The source for for [data_source_variables ](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/data_source_variables.go) seems to lack the patterns/conventions used in other data sources in this provider:
- non-Computed parameters used for querying ([example](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/schema_environment.go#L62-L64)) 
- Flattening Octopus API results back into the output data ([example](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/schema_environment.go#L35))

**Steps to reproduce**
1. Use a `octopusdeploy_variables` block in your terraform file as per https://registry.terraform.io/providers/OctopusDeployLabs/octopusdeploy/latest/docs/data-sources/variables
2. Note that you cannot set any parameters for querying for variables, as none are defined in the schema: https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/schema_variable.go#L66-L68
3. When Terraform executes the `octopusdeploy_variables` block, it attempts to read in values from fields that are not defined in the schema and/or are `Computed: true`. https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/data_source_variables.go#L20-L21

**Expected behavior**
Should be able to read Octopus variables from an Octopus Server installation and query by name, variable set id, etc.

**Logs and other supporting information**
- When no parameters supplied to `octopusdeploy_variables` block:
```
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-octopusdeploy_v0.7.69.exe plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 30 [running]:
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.dataSourceVariableReadByName({0x1044098, 0xc00007ca00}, 0x19a97ac1c68, {0xda5340, 0xc0000d9180})
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/data_source_variables.go:29 +0x577
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00026d5e0, {0x1044098, 0xc00007ca00}, 0xc0001a4300, {0xda5340, 0xc0000d9180})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:358 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc00026d5e0, {0x1044098, 0xc00007ca00}, 0xc000548080, {0xda5340, 0xc0000d9180})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:569 +0xf7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000696048, {0x1044098, 0xc00007ca00}, 0xc00020c1c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:1133 +0x3a8
```

- When `owner_id` is supplied to the block:
```
│ Error: Unsupported argument
│
│   on foo.tf line 16, in data "octopusdeploy_variables" "myvars":
│   16:   owner_id = "test-data"
│
│ An argument named "owner_id" is not expected here.
```


**Additional context**
When I use a locally modified version of the provider and expand the `data_source_variables` file by copying patterns/conventions used in the other data sources I am able to get the `octopusdeploy_variables` data source to function as I expect. I am wondering if the implementation for this particular data source was just missed or if I am not understanding/using it correctly.

--- a/octopusdeploy/data_source_variables.go
+++ b/octopusdeploy/data_source_variables.go
@@ -18,36 +18,29 @@ func dataSourceVariable() *schema.Resource {
 }
 
 func dataSourceVariableReadByName(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	ownerID := d.Get("owner_id")
-	name := d.Get("name")
-	scope := variables.VariableScope{}
+	ID := d.Get("id").(string)
+	spaceID := d.Get("space_id").(string)
 
-	if v, ok := d.GetOk("scope"); ok {
-		scope = expandVariableScope(v)
-	}
+	ownerID := d.Get("owner_id").(string)
 
-	var spaceID string
-	if v, ok := d.GetOk("space_id"); ok {
-		spaceID = v.(string)
-	}
+	name := d.Get("name").(string)
+
+	scope := expandVariableScope(d.Get("scope"))
 
 	client := m.(*client.Client)
-	variables, err := variables.GetByName(client, spaceID, ownerID.(string), name.(string), &scope)
+	variables, err := variables.GetByName(client, spaceID, ownerID, name, &scope)
 	if err != nil {
-		return diag.Errorf("error reading variable with owner ID %s with name %s: %s", ownerID, name, err.Error())
+		return diag.Errorf("error reading variable with owner ID %s with ID %s: %s", ownerID, ID, err.Error())
 	}
 	if variables == nil {
 		return nil
 	}
-	if len(variables) > 1 {
-		return diag.Errorf("found %v variables with owner ID %s with name %s, should match exactly 1", len(variables), ownerID, name)
+
+	if len(variables) != 1 {
+		return diag.Errorf("error could not find variable by name, expected to find 1 variable but got %d", len(variables))
 	}
 
-	d.SetId(variables[0].ID)
-	d.Set("name", variables[0].Name)
-	d.Set("type", variables[0].Type)
-	d.Set("value", variables[0].Value)
-	d.Set("description", variables[0].Description)
+	setVariable(ctx, d, variables[0])
 
 	return nil
 }

--- a/octopusdeploy/data_source_variables.go
+++ b/octopusdeploy/data_source_variables.go
@@ -18,7 +18,6 @@ func dataSourceVariable() *schema.Resource {
 }
 
 func dataSourceVariableReadByName(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	ID := d.Get("id").(string)
 	spaceID := d.Get("space_id").(string)
 
 	ownerID := d.Get("owner_id").(string)
@@ -30,7 +29,7 @@ func dataSourceVariableReadByName(ctx context.Context, d *schema.ResourceData, m
 	client := m.(*client.Client)
 	variables, err := variables.GetByName(client, spaceID, ownerID, name, &scope)
 	if err != nil {
-		return diag.Errorf("error reading variable with owner ID %s with ID %s: %s", ownerID, ID, err.Error())
+		return diag.Errorf("error reading variable with owner ID %s with name %s: %s", ownerID, name, err.Error())
 	}
 	if variables == nil {
 		return nil

--- a/octopusdeploy/schema_variable.go
+++ b/octopusdeploy/schema_variable.go
@@ -55,19 +55,107 @@ func expandVariable(d *schema.ResourceData) *variables.Variable {
 }
 
 func getVariableDataSchema() map[string]*schema.Schema {
-	dataSchema := getVariableSchema()
-	setDataSchema(&dataSchema)
-
 	return map[string]*schema.Schema{
-		"id":       getDataSchemaID(),
-		"space_id": getQuerySpaceID(),
-		"ids":      getQueryIDs(),
-		"variables": {
+		"id": {
 			Computed:    true,
-			Description: "A list of variables that match the filter(s).",
-			Elem:        &schema.Resource{Schema: dataSchema},
-			Optional:    true,
+			Description: "The identifier of the variable to find.",
+			Type:        schema.TypeString,
+		},
+		"name": {
+			Required:    true,
+			Description: "The name of variable to find.",
+			Type:        schema.TypeString,
+		},
+		"owner_id": {
+			Required:    true,
+			Description: "Owner ID for the variable to find.",
+			Type:        schema.TypeString,
+		},
+		"scope": {
+			Description: "As variable names can appear more than once under different scopes, a VariableScope must also be provided",
+			Elem:        &schema.Resource{Schema: getVariableScopeSchema()},
+			Required:    true,
 			Type:        schema.TypeList,
+			MaxItems:    1,
+		},
+		"space_id": getQuerySpaceID(),
+		"description": {
+			Description: "The description of this variable.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"is_editable": {
+			Description: "Indicates whether or not this variable is considered editable.",
+			Computed:    true,
+			Type:        schema.TypeBool,
+		},
+		"is_sensitive": {
+			Description: "Indicates whether or not this resource is considered sensitive and should be kept secret.",
+			Computed:    true,
+			Type:        schema.TypeBool,
+		},
+		"type": {
+			Description: "The type of variable represented by this resource. Valid types are `AmazonWebServicesAccount`, `AzureAccount`, `GoogleCloudAccount`, `Certificate`, `Sensitive`, `String`, or `WorkerPool`.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"sensitive_value": {
+			Computed:  true,
+			Sensitive: true,
+			Type:      schema.TypeString,
+		},
+		"value": {
+			Computed: true,
+			Type:     schema.TypeString,
+		},
+		"prompt": {
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"description": getDescriptionSchema("variable prompt option"),
+					"display_settings": {
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"control_type": {
+									Description: "The type of control for rendering this prompted variable. Valid types are `SingleLineText`, `MultiLineText`, `Checkbox`, `Select`.",
+									Required:    true,
+									Type:        schema.TypeString,
+								},
+								"select_option": {
+									Elem: &schema.Resource{
+										Schema: map[string]*schema.Schema{
+											"value": {
+												Description: "The select value",
+												Required:    true,
+												Type:        schema.TypeString,
+											},
+											"display_name": {
+												Description: "The display name for the select value",
+												Required:    true,
+												Type:        schema.TypeString,
+											},
+										},
+									},
+									Description: "If the `control_type` is `Select`, then this value defines an option.",
+									Optional:    true,
+									Type:        schema.TypeList,
+								},
+							},
+						},
+						Optional: true,
+						Type:     schema.TypeList,
+					},
+					"is_required": {
+						Type:     schema.TypeBool,
+						Optional: true,
+					},
+					"label": {
+						Type:     schema.TypeString,
+						Optional: true,
+					},
+				},
+			},
+			Computed: true,
+			Type:     schema.TypeList,
 		},
 	}
 }

